components/scream/cmake/ScreamUtils.cmake

```diff
@@ -11,15 +11,20 @@ include(CMakeParseArguments) # Needed for backwards compatibility
 #      Note: One test will be created per combination of valid mpi-rank and thread value
 
 FUNCTION(CreateUnitTest target_name target_srcs scream_libs)
+  set(options OPTIONAL EXCLUDE_MAIN_CPP)
   set(oneValueArgs NAMELIST_FILE)
   set(multiValueArgs MPI_RANKS THREADS CONFIG_DEFS INCLUDE_DIRS)
-  cmake_parse_arguments(CreateUnitTest "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
+  cmake_parse_arguments(CreateUnitTest "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
 
   # Set link directories (must be done BEFORE add_executable is called)
   link_directories(${SCREAM_TPL_LIBRARY_DIRS})
 
   # Create the executable
-  add_executable (${target_name} ${target_srcs} ${SCREAM_SRC_DIR}/share/util/scream_catch_main.cpp)
+  set (SRC_MAIN ${SCREAM_SRC_DIR}/share/util/scream_catch_main.cpp)
+  if (CreateUnitTest_EXCLUDE_MAIN_CPP)
+    set (SRC_MAIN)
+  endif ()
+  add_executable (${target_name} ${target_srcs} ${SRC_MAIN})
```

Comment:
Permit creating a main while also using the catch2 framework established in this function.